### PR TITLE
[MM-13500] Add search group channels method to client

### PR DESCRIPTION
**Pull request by mgdelacroix**
#### Ticket Link
https://mattermost.atlassian.net/browse/MM-13500

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit tests passed
- [x] Ran `make flow` to ensure type checking passed
- [x] All new/modified APIs include changes to the drivers

### CODE REVIEW

**Comment by jasonblais** [comment]
@mgdelacroix Looks like this one is ready to merge

**Comment by mgdelacroix** [comment]
@jasonblais it is dependant on the changes of the related PRs, I'm working on updating those

File: src/actions/users.js [hunk 1]
```diff
@@ -504,6 +504,44 @@ export function getProfilesInChannel(channelId: string, page: number, perPage: n
...
+                    },
+                    {
+                        type: UserTypes.RECEIVED_PROFILES_LIST,
+                        data: removeUserFromList(currentUserId, [...profiles]),
+                    },
+                );
+            }
+        }
+
+        dispatch(batchActions(actions), getState);
```
Comment (jwilander) [inline src/actions/users.js]: No need to pass `getState` to dispatch

```suggestion
        dispatch(batchActions(actions));
```

Comment (mgdelacroix) [reply src/actions/users.js]: Done!

**Pull request merged**